client/os.cpp

```diff
@@ -260,6 +263,10 @@ namespace tec {
 	void OS::OSMessageLoop() {
 		glfwPollEvents();
 		EventQueue<KeyboardEvent>::ProcessEventQueue();
+		if (this->mouse_lock != OS::mouse_locked) {
```

Comment:
Is there a reason to do this check every frame instead of via the enable/disable methods?

Reply:
The enable/disable methods could be called from any thread, but the `glfwSetInputMode` call has to be on this thread or it doesn't work.